test/jdk/javax/swing/JColorChooser/Test6827032.java

```diff
@@ -38,47 +38,57 @@
 
 
 public class Test6827032 {
-
+    private static JFrame frame;
     private static volatile Point point;
     private static JColorChooser cc;
 
     public static void main(String[] args) throws Exception {
-        UIManager.setLookAndFeel(new NimbusLookAndFeel());
-
-        Robot robot = new Robot();
-        robot.setAutoDelay(50);
-
-
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                createAndShowGUI();
+        try {
+            UIManager.setLookAndFeel(new NimbusLookAndFeel());
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+
+            SwingUtilities.invokeAndWait(new Runnable() {
+                public void run() {
+                    createAndShowGUI();
+                }
+            });
+
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(new Runnable() {
+                public void run() {
+                    Component previewPanel = Util.findSubComponent(cc, "javax.swing.colorchooser.DefaultPreviewPanel");
+                    point = previewPanel.getLocationOnScreen();
+                }
+            });
+
+            point.translate(5, 5);
+
+            robot.mouseMove(point.x, point.y);
+            robot.waitForIdle();
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(1000);
+        } finally {
+            if (frame != null) {
+                SwingUtilities.invokeAndWait(() -> frame.dispose());
```

Comment:
Shall the frame be `volatile` too? It's accessed from main thread for the null-check.